### PR TITLE
Adds milestone edit modal

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/9667

## Testing
**URL to test:** 
https://deploy-preview-769--atd-moped-main.netlify.app/

**Steps to test:**
go to dashboard and click the status badge or milestones meter for a modal to update phases or milestones, respectively

there is a minor issue where there is a delay from when the modal is opened to when the table is displayed. fixing this will require refactoring the timeline component. i will make an issue to include this refactor in an upcoming release.

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/dashboard/DashboardTimelineModal.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        {contents}
+      </div>
+      <Dialog
+        open={isDialogOpen}
+        onClose={handleDialogClose}
+        fullWidth
+        maxWidth={"xl"}
+      >
+        <DialogTitle disableTypography className={classes.dialogTitle}>
+          <h4>{`Update phase - ${projectName}`}</h4>
```
Comment (chiaberry) [inline moped-editor/src/views/dashboard/DashboardTimelineModal.js:55]: This needs to toggle based on the table name

File: moped-editor/src/views/dashboard/DashboardTimelineModal.js [hunk 2]
```diff
@@ -53,17 +52,20 @@ const DashboardPhaseModal = ({
         maxWidth={"xl"}
       >
         <DialogTitle disableTypography className={classes.dialogTitle}>
-          <h4>{`Update phase - ${project.project_name}`}</h4>
+          <h4>{`Update phase - ${projectName}`}</h4>
```
Comment (chiaberry) [inline moped-editor/src/views/dashboard/DashboardTimelineModal.js:55]: This needs to toggle based on the table name

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js [hunk 3]
```diff
@@ -496,242 +496,244 @@ const ProjectTimeline = ({
...
-                    // If one of the Date fields is blanked out, coerce empty
-                    // string to null.
-                    differences.forEach((diff) => {
-                      let shouldCoerceEmptyStringToNull =
-                        newData[diff] === "" &&
-                        (diff === "phase_start" || diff === "phase_end");
-
-                      if (shouldCoerceEmptyStringToNull) {
-                        updatedPhaseObject[diff] = null;
+          {table !== "milestones" && (
```
Comment (chiaberry) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:499]: github is struggling a bit with the diff, but as far as I can tell the only changes (besides indentation) from here down is the conditional rendering based on table name, right?

**Comment by chiaberry** [comment]
Github was showing the incorrect diffs, so I merged main to it and pushed the branch again to fix the diffs so if anyone was reviewing today / monday.

**Review by chiaberry — CHANGES_REQUESTED** [review]
Found a spot that needs to toggle based on table name but otherwise looks good!

**Review by mateoclarke — APPROVED** [review]
🚢💨✨

**Review by amenity — APPROVED** [review]
Works well! Thanks, @tillyw.

File: moped-editor/src/views/dashboard/DashboardTimelineModal.js (hunk 2 above)
Comment (tillyw) [reply moped-editor/src/views/dashboard/DashboardTimelineModal.js:55]: Thank you for catching this!! Fixed

**Comment by tillyw** [comment]
> Github was showing the incorrect diffs, so I merged main to it and pushed the branch again to fix the diffs so if anyone was reviewing today / monday.

sorry I forgot to pull and merge main before I submitted this pr, thanks for fixing that!

**Pull request merged**